2-collectors/thrift-schemas/collector-payload-1/src/main/thrift/collector-payload.thrift

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2014 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0, and
+ * you may not use this file except in compliance with the Apache License
+ * Version 2.0.  You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the Apache License Version 2.0 is distributed on an "AS
+ * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+
+namespace java com.snowplowanalytics.snowplow.CollectorPayload.thrift.model1
+
+struct CollectorPayload {
+  31337: string schema
+  10: optional string querystring
+  20: string collector
+  30: string encoding
+  40: optional string hostname
+  50: i64 timestamp
+  60: string ipAddress
+  70: optional string userAgent
+  80: optional string refererUri
+  90: optional list<string> headers
+  100: optional string networkUserId
+  110: string path
+  120: optional string contentType
+  130: optional string body
+}
```

Comment:
I see what you mean. I've made a new branch with a hopefully more readable version [here](https://github.com/snowplow/snowplow/commit/5fdbd597cb1954dd6a4e9049f259bb81235e3071).

It looks like the AdapterRegistry rejects anything without an approved path [here](https://github.com/snowplow/snowplow/blob/master/3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/AdapterRegistry.scala#L77) - wouldn't this reject all events from the Clojure and Cloudfront Collectors with a null path?